test/realtime/resume.test.js

```diff
@@ -629,5 +624,67 @@ define(['shared_helper', 'async', 'chai'], function (helper, async, chai) {
         closeAndFinish(done, [sender_realtime, receiver_realtime, resumed_receiver_realtime], err);
       }
     });
+
+    it('recover multiple channels', async function () {
+      const NUM_MSGS = 5;
+
+      const txRest = helper.AblyRest();
+      const rxRealtime = helper.AblyRealtime({ transports: [helper.bestTransport] }, true);
+
+      const channelNames = Array(5).fill().map(() => String(Math.random()));
+      const rxChannels = channelNames.map((name) => rxRealtime.channels.get(name));
+
+      await Promise.all(rxChannels.map((channel) => channel.attach()));
+
+      // Do a few publishes on each channel so the channelSerial is set.
+      await Promise.all(channelNames.map(async (name) => {
+        const tx = txRest.channels.get(name);
+        const rx = rxRealtime.channels.get(name);
+        await rx.attach();
+
+        for (let i = 0; i < NUM_MSGS; i++) {
+          const pSubscribe = rx.subscriptions.once();
+          await tx.publish(null, null);
+          await pSubscribe;
+        }
+      }));
+
+      // Get the recovery key before becoming suspended as it will be reset.
+      const recoveryKey = rxRealtime.connection.getRecoveryKey();
+
+      await new Promise((resolve) => helper.becomeSuspended(rxRealtime, resolve));
+
+      // Send some messages after we've detached that should be recovered.
+      for (const name of channelNames) {
+        const tx = txRest.channels.get(name);
+        for (let i = 0; i < NUM_MSGS; i++) {
+          await tx.publish('sentWhileDisconnected', null);
+        }
+      }
+
+      const rxRealtimeRecover = helper.AblyRealtime({ recover: recoveryKey });
+      const rxRecoverChannels = channelNames.map((name) => rxRealtimeRecover.channels.get(name));
+
+      // Attach recovered channels and check we receive the expected messages.
+      await Promise.all(rxRecoverChannels.map(async (channel) => {
+        await channel.attach();
+
+        await new Promise((resolve) => {
```

Comment:
I see two issues with this approach
1. Risk of test waiting for an indefinite amount of time to match the assertion, so it either passes or goes into wait state 
( rather than failing )
2. Where are we passing exception thrown by `expect` statement. As far as I know, thing work differently when you put `expect` statements inside a method. 
So, we need a separate method that does that, the `new Promise` would look like

```
const safeAssert = async (assertions: Function) => {
await new Promise((resolve, reject) => {
    try {
        assertions();
        resolve();
    } catch (err) {
        reject(err)
    }
});
});
await safeAssert(()=> {
          let recoveredCount = 0;
          channel.subscribe((msg) => {
            // Check we don't receive unexpected messages.
            expect(msg.name).to.equal('sentWhileDisconnected');

            recoveredCount++;
            if (recoveredCount === NUM_MSGS) {
              resolve();
            }
          });
})

```
This fix still doesn't cover first problem.